FreeRTOS-Plus/Demo/FreeRTOS_Cellular_Interface_Windows_Simulator/Common/cellular_setup.c

```diff
@@ -131,27 +127,55 @@ bool setupCellular( void )
 
             vTaskDelay( pdMS_TO_TICKS( CELLULAR_SIM_CARD_WAIT_INTERVAL_MS ) );
         }
+
+        if( cellularStatus != CELLULAR_SUCCESS )
+        {
+            configPRINTF( ( ">>>  Cellular SIM failure  <<<\r\n" ) );
+        }
+    }
+
+    /* Turn off PSM because this is demo to showcase MQTT instead of PSM mode. */
+    if( cellularStatus != CELLULAR_SUCCESS )
+    {
+        psmSettings.mode = 0;
+        cellularStatus = cellularStatus = Cellular_SetPsmSettings( CellularHandle, &psmSettings );
+
+        if( cellularStatus != CELLULAR_SUCCESS )
+        {
+            configPRINTF( ( ">>>  Cellular_SetPsmSettings failure  <<<\r\n" ) );
+        }
     }
 
     /* Setup the PDN config. */
     if( cellularStatus == CELLULAR_SUCCESS )
     {
         cellularStatus = Cellular_SetPdnConfig( CellularHandle, CellularSocketPdnContextId, &pdnConfig );
-    }
-    else
-    {
-        configPRINTF( ( ">>>  Cellular SIM failure  <<<\r\n" ) );
+
```

Comment:
Shouldn't PDN Configuration happen after registration? (with the exception of the default bearer config).

Reply:
Yes, this step setup the default bearer config.
Cellular modem can not register to the network if APN setting is wrong for default bearer.
Network can provide the APN when the PLMN is manually assigned by user.
Currently, cellular interface library use automatic registration by modem.